docs/dev/release.md

```diff
@@ -1,62 +1,64 @@
 
-```shell
-# Tag the commit with version info
-git tag {version}  # where version is a version number like "1.7.0"
+- Create a new release notes file for the next bugfix release.
+  This should be included in both dev and stable branches.
+- Create a milestone for the next release (in case you made a bugfix release) or releases (in case of a major/minor release).
+  For bugfix releases, this should have `on-merge: backport to 0.<minor>.x`,
+  so the [meeseeksdev][] bot will create a backport PR. See {doc}`versioning` for more info.
+- Clear out and close the milestone you just made a release for.
 
-# Clear out old distributions
-rm -r dist
+After a *major* or *minor* release has been made:
 
-# Build distributions and wheel
-python -m build
-# you can check that the previous step worked by installing from it's results
-# e.g. pip install dist/scanpy-{version}-py3-none-any.whl
+- Tweet about it! Announce it on Zulip! Announce it on Discourse! Think about making a bot for this! Maybe actually do that?
+- Create a new release notes file for the next minor release. This should only be added to the dev branch.
+- Tag the development branch. If you just released `1.7.0`, this would be `1.8.0.dev0`.
+- Create a new branch for this release series, like `1.7.x`. This should get a new release notes file.
 
-# Once you're confident the build looks good, push the tag to github
-git push upstream {version}
+[meeseeksdev]: https://meeseeksbox.github.io
 
-# Upload wheel and code distribution to pypi
-twine upload dist/scanpy-{version}
-```
+## Debugging the build process
 
-### Checking the distributions
+If you changed something about the build process (e.g. [Hatchling’s build configuration][hatch-build]),
+or something about the package’s structure,
+you might want to manually check if the build and upload process behaves as expected:
 
-If you're feeling cautious, you can:
+```shell
+# Clear out old distributions
+rm -r dist
 
-- List the contents of a wheel with: `unzip -l dist/*.whl`
-- Make a release candidate. Tag these with `{version}rc1` (e.g. `1.7.0rc1`)
-- Upload them to <test.pypi.org> ([tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/#uploading-the-distribution-archives))
+# Build source distribution and wheel both
+python -m build
 
-## After making a release
+# Now check those build artifacts
+twine check dist/*
 
-After a major or minor release has been made:
+# List the wheel archive’s contents
+bsdtar -tf dist/*.whl
 
-- Tweet about it! Announce it on Zulip! Announce it on Discourse! Think about making a bot for this! Maybe actually do that?
-- Create a new release notes file for the next minor release. This should only be added to the dev branch.
-- Tag the development branch. If you just released `1.7.0`, this would be `1.8.0.dev0`.
-- Create a new branch for this release series, like `1.7.x`. This should get a new release notes file.
+# Upload to TestPyPI
+twine upload --repository testpypi dist/*
+```
```

Comment:
````suggestion
```

You can also upload the package to <test.pypi.org> ([tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/#uploading-the-distribution-archives))

```
twine upload --repository testpypi dist/*
```
````

I'd like to keep the link here. There are more steps to using test pypi (e.g. make an account etc, point pip at it to download result).

Reply:
sure. Please don‘t do inline links though